idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2296 @@
+    "    m.fs.lex.aqueous_inlet.conc_mass_comp[0, \"CaSO4\"].fix(0.1 * pyo.units.g / pyo.units.L)\n",
+    "\n",
+    "    return m \n",
+    "\n",
+    "def initialize_model(m):\n",
+    "    initializer = BlockTriangularizationInitializer()\n",
+    "    initializer.initialize(m.fs.lex)\n",
+    "    return m\n",
+    "\n",
+    "def main():\n",
+    "    m=build_model()\n",
+    "    m=fix_state_variables(m)\n",
+    "    m=initialize_model(m)\n",
+    "    return m\n",
+    "\n",
+    "if __name__==main:\n",
+    "    main()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7971e22d",
+   "metadata": {},
+   "source": [
+    "# 4. Model Diagnostics using DiagnosticsToolbox\n",
+    "\n",
+    "Here, during initialization, we encounter warnings indicating that variables are being set to negative values, which is not expected behavior. These warnings suggest that there may be flaws in the model that require further investigation using the DiagnosticsToolbox from IDAES. A detailed notebook on using `DiagnosticsToolbox` can be found [here](../../diagnostics/degeneracy_hunter.ipynb).\n",
+    "\n",
+    "To proceed with investigating these issues, we need to import the DiagnosticsToolbox. We can gain a better understanding of its functionality by running the help function on it. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "27f7d6c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from idaes.core.util import DiagnosticsToolbox"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe1f5a62",
+   "metadata": {},
+   "source": [
+    "The help() function provides comprehensive information on the DiagnosticsToolbox and all its supported methods. However, it's essential to focus on the initial steps outlined at the beginning of the docstring to get started effectively.\n",
+    "\n",
+    "Here's a breakdown of the steps to start with:\n",
+    "\n",
+    "- `Instantiate Model:` Ensure you have an instance of the model with a degrees of freedom equal to 0.\n",
+    "\n",
+    "- `Create DiagnosticsToolbox Instance:` Next, instantiate a DiagnosticsToolbox object.\n",
+    "\n",
+    "- `Provide Model to DiagnosticsToolbox:` Pass the model instance to the DiagnosticsToolbox.\n",
+    "\n",
+    "- `Call report_structural_issues() Function:` Finally, call the report_structural_issues() function. This function will highlight any warnings in the model's structure, such as unit inconsistencies or other issues related to variables in the caution section.\n",
+    "\n",
+    "By following these steps, you can efficiently utilize the DiagnosticsToolbox to identify and address any structural issues or warnings in your model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "307ec415",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WARNING (W1001): Setting Var\n",
```

Comment:
These warnings should be fixed if possible - this suggests you are setting some values to 0 when the bound is a small positive number.

Reply:
This warnings are for demonstration purposes. I resolve them later with diagnostics. It is caused because there are numerical issues with the model.